.github/workflows/test-all.yml

```diff
@@ -196,8 +183,28 @@ jobs:
             echo "No required binary depedencies to be installed"
           fi
 
+      - name: "Check if tests should be skipped"
+        id: checks
+        run: |
+          PKG=${{ matrix.package }}
+          if ! jq -e -r '.TestFile' < packages/${PKG}/meta.json > /dev/null ; then
```

Comment:
Hrm. But this means launching ~20 additional CI tests (each taking 30-60 seconds, and downloading a couple hundred MB) just to confirm that nothing is going to happen.

Can't we instead collect all this info in the initial step, and put it into one JSON artifact, which then is merged together with the others? Or perhaps even simpler, generate multiple JSON files in the "usual" format, and upload them in one artifact, then add them to the other such JSON files we collect from the "real" CI jobs.